tests/test_module/test_rest_node.py

```diff
@@ -2,235 +2,88 @@
-
-
-@app.post("/action")
-def do_action(action_handle: str, action_vars: str, files: List[UploadFile] = []):  # noqa: B006
-    """
-    Runs an action on the module
-
-    Parameters
-    ----------
-    action_handle : str
-       The name of the action to be performed
-    action_vars : str
-        Any arguments necessary to run that action.
-        This should be a JSON object encoded as a string.
-    files: List[UploadFile] = []
-        Any files necessary to run the action defined by action_handle.
-
-    Returns
-    -------
-    response: StepResponse
-       A response object containing the result of the action
-    """
-    global state, foobar
-    if state == ModuleStatus.BUSY:
-        return StepResponse(
-            action_response=StepStatus.Failed,
-            action_msg="",
-            action_log="Node is busy",
-        )
-    state = ModuleStatus.BUSY
-    try:
-        if action_handle == "synthesize":
-            action_vars = json.loads(action_vars)
-            foo = float(action_vars["foo"])
-            bar = float(action_vars["bar"])
-            protocol = next(file for file in files if file.filename == "protocol")
-            protocol = protocol.file.read().decode("utf-8")
-            print(protocol)
-
-            foobar = foo + bar
-
-            state = ModuleStatus.IDLE
-            return StepResponse(
-                action_response=StepStatus.SUCCEEDED,
-                action_msg="",
-                action_log="Synthesized sample {foo} + {bar}",
-            )
-        elif action_handle == "measure":
-            action_vars = json.loads(action_vars)
-
-            state = ModuleStatus.IDLE
-            return StepResponse(
-                action_response=StepStatus.SUCCEEDED,
-                action_msg=str(foobar),
-                action_log="",
-            )
-        elif action_handle == "transfer":
-            action_vars = json.loads(action_vars)
-            source = action_vars.get("source", None)
-            target = action_vars["target"]
-
-            state = ModuleStatus.IDLE
-            return StepResponse(
-                action_response=StepStatus.SUCCEEDED,
-                action_msg="",
-                action_log=f"Moved sample from {source} to {target}",
-            )
-        else:
-            state = ModuleStatus.IDLE
-            return StepResponse(
-                action_response=StepStatus.FAILED,
-                action_msg="False",
-                action_log=f"Action {action_handle} not supported",
-            )
-    except Exception as e:
-        print(e)
-        state = ModuleStatus.ERROR
-        return StepResponse(
-            action_response=StepStatus.FAILED,
-            action_msg="False",
-            action_log=str(e),
-        )
-
-
-if __name__ == "__main__":
-    import uvicorn
-
-    uvicorn.run(
-        "test_rest_node:app",
-        host=args.host,
-        port=args.port,
-        reload=True,
-    )
+test_rest_node.start()
```

Comment:
Might be good to add:

`if __name__ == "__main__":`
`    test_rest_node.start()`